integrations/aws/utils.py

```diff
@@ -0,0 +1,309 @@
+import enum
+import json
+from typing import Any, AsyncIterator, Optional
+import typing
+import aioboto3
+from loguru import logger
+from overrides import AWSPortAppConfig
+from port_ocean.context.event import event
+from port_ocean.context.ocean import ocean
+from starlette.requests import Request
+from port_ocean.core.ocean_types import ASYNC_GENERATOR_RESYNC_TYPE
+from port_ocean.core.handlers.port_app_config.models import ResourceConfig
+
+ACCOUNT_ID_PROPERTY = '__AccountId'
+KIND_PROPERTY = '__Kind'
+REGION_PROPERTY = '__Region'
+
+class ResourceKindsWithSpecialHandling(enum.StrEnum):
+    """
+    Resource kinds with special handling
+    These resource kinds are handled separately from the other resource kinds
+    """
+    ACCOUNT = "AWS::Organizations::Account"
+    CLOUDRESOURCE = "cloudresource"
+    EC2 = "AWS::EC2::Instance"
+    CLOUDFORMATION = "AWS::CloudFormation::Stack"
+    LOADBALANCER = "AWS::ElasticLoadBalancingV2::LoadBalancer"
+    ACM = "AWS::ACMPCA::Certificate"
+
+class AwsCredentials:
+    def __init__(self, account_id: str, access_key_id: str, secret_access_key: str, session_token: Optional[str] = None):
+        self.account_id = account_id
+        self.access_key_id = access_key_id
+        self.secret_access_key = secret_access_key
+        self.session_token = session_token
+        self.enabled_regions = []
+    
+    async def updateEnabledRegions(self):
+        session = aioboto3.Session(self.access_key_id, self.secret_access_key, self.session_token)
+        async with session.client("account") as account_client:
+            response = await account_client.list_regions(RegionOptStatusContains=['ENABLED', 'ENABLED_BY_DEFAULT'])
+            regions = response.get("Regions", [])
+            self.enabled_regions = [region["RegionName"] for region in regions]
+
+    def isRole(self):
+        return self.session_token is not None
+    
+    async def createSession(self, region: str) -> aioboto3.Session:
+        if self.isRole():
+            return aioboto3.Session(self.access_key_id, self.secret_access_key, self.session_token, region)
+        else:
+            return aioboto3.Session(aws_access_key_id=self.access_key_id, aws_secret_access_key=self.secret_access_key, region_name=region)
```

Comment:
why dont you work only with the role?

Reply:
I could, but running this integration `on-prem` might not use a cli